tools/eslint-rules/restrict-let-in-for-loops.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Prohibit the use of `let` in `for` loops.
+ * @author Jessica Quynh Tran
+ */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  create(context) {
+
+    const msg = 'Use of `let` declaration within a for-loop is ' +
+                'not recommended. Please use `var` instead.';
+
+    function testForLoop(node) {
+      return node.init && node.init.kind === 'let';
+    }
+
+    function testForInOfLoop(node) {
+      return node.left && node.left.kind === 'let';
+    }
+
+    /**
+     * Report function to test if the for-loop is declared using `let`.
+     */
+    function checkForLet(node) {
+      if (testForLoop(node) || testForInOfLoop(node)) {
```

Comment:
That makes a lot of sense! I will update the test and add it to the .eslintrc. I was wondering if it would be more appropriate to report an error or simply a warning on the rule?